hips/draw/simple.py

```diff
@@ -83,111 +87,60 @@ def dst(self) -> np.ndarray:
-        A list of HEALPix pixel indices
-    order : int
-        Order of the HEALPix map
-    hips_survey : HipsSurveyProperties
-        An object of HipsSurveyProperties
-    tile_format : `str`
-        Format of HiPS tile
-
-    Returns
-    -------
-    'HipsTile'
-        Returns an object of HipsTile
-    """
-    for healpix_pixel_index in healpix_pixel_indices:
-        tile_meta = HipsTileMeta(
-            order=order,
-            ipix=healpix_pixel_index,
-            frame=hips_survey.astropy_frame,
-            file_format=tile_format,
-        )
-        tile = HipsTile.fetch(tile_meta, hips_survey.tile_access_url(order=order, ipix=healpix_pixel_index) + tile_meta.filename)
-        yield tile
-
-
-def compute_matching_hips_order(geometry: WCSGeometry, hips_survey: HipsSurveyProperties) -> int:
-    """Compute HiPS tile order matching a given image pixel size.
-
-    Parameters
-    ----------
-    geometry : WCSGeometry
-        Geometry of the output image
-    hips_survey : HipsSurveyProperties
-        An object of HipsSurveyProperties
-
-    Returns
-    -------
-    'int'
-        Returns HiPS order
-
-    Examples
-    --------
-    >>> from hips.draw import compute_matching_hips_order
-    >>> from astropy.coordinates import SkyCoord
-    >>> url = 'http://alasky.unistra.fr/DSS/DSS2Merged/properties'
-    >>> hips_survey = HipsSurveyProperties.fetch(url)
-    >>> geometry = WCSGeometry.create_simple(
-    ...     skydir=SkyCoord(0, 0, unit='deg', frame='icrs'),
-    ...     width=2000, height=1000, fov="3 deg",
-    ...     coordsys='icrs', projection='AIT'
-    ... )
-    >>> compute_matching_hips_order(geometry, hips_survey)
-    7
-    """
-
-    # Sky image angular resolution (pixel size in degree)
-    resolution = np.min(proj_plane_pixel_scales(geometry.wcs))
-    desired_order = _get_hips_order_for_resolution(hips_survey.tile_width, resolution)
-    # Return the desired order, or the highest resolution available.
-    # Note that HiPS never has resolution less than 3,
-    # and that limit is handled in _get_hips_order_for_resolution
-    return np.min([desired_order, hips_survey.hips_order])
-
-
-def _get_hips_order_for_resolution(tile_width, resolution):
-    """Finding the best HiPS order by looping through all possible options."""
-    tile_order = np.log2(tile_width)
-    full_sphere_area = 4 * np.pi * np.square(180 / np.pi)
-    # 29 is the maximum order supported by healpy and 3 is the minimum order
-    for candidate_tile_order in range(3, 29 + 1):
-        tile_resolution = np.sqrt(full_sphere_area / 12 / 4 ** (candidate_tile_order + tile_order))
-        # Finding the smaller tile order with a resolution equal or better than geometric resolution
-        if tile_resolution <= resolution:
-            break
-
-    return candidate_tile_order
+    def run(self) -> None:
+        """Run all steps of naive algorithm"""
+        self.tiles = self.fetch_tiles()
+        self.image = np.zeros(self.shape)
```

Comment:
This should be changed to `self.image = self.draw_tiles()`, i.e. move those lines doing the drawing and stitching of the tiles to a method that can be executed and tested separately.